src/linear_algebra/matrix/charpoly/eigs.lean

```diff
@@ -0,0 +1,59 @@
+/-
+Copyright (c) 2023 Mohanad Ahmed. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mohanad Ahmed
+-/
+
+import data.polynomial.basic
+import field_theory.is_alg_closed.basic
+import algebra.char_p.two
+
+/-!
+# Eigenvalues are characteristic polynomial roots.
+
+In algebraically closed fields, we show:
+
+* `matrix.det_eq_prod_roots_charpoly`: the determinant is the product of the roots of the
+  characteristic polynomial.
+* `matrix.trace_eq_sum_roots_charpoly`: the trace is the sum of the roots of the characteristic
+  polynomial.
+
+Note that over other fields such as ` ℝ`, these results can be used by using
+`A.map (algebra_map ℝ ℂ)` as the matrix, and then applying `ring_hom.map_det`.
+
+The two lemmas `matrix.det_eq_prod_roots_charpoly` and `matrix.trace_eq_sum_roots_charpoly` are more
+commonly stated as trace is the sum of eigenvalues and determinant is the product of eigenvalues.
+Mathlib has already defined eigenvalues in `linear_algebra.eigenspace` as the roots of the minimal
+polynomial of a linear endomorphism. These do not have correct multiplicity and cannot be used in
+the theorems above. Hence we express these theorems in terms of the roots of the characteristic
+polynomial directly.
+-/
+variables {n : Type*} [fintype n] [decidable_eq n]
+variables {R : Type*} [field R] [is_alg_closed R]
```

Comment:
We only really need that one specific polynomial splits, so let's weaken this hypothesis.
```suggestion
variables {R : Type*} [field R] (h : (matrix.charpoly A).splits)
```

Reply:
This seems to require and extra argument `(i : K →+* L)`. It says this "bundled ring homomorphisms".[ring_hom](https://leanprover-community.github.io/mathlib_docs/algebra/hom/ring.html#ring_hom).
I have no idea what that is or how to get one. As you can guess my lack of knowledge about this thing, makes me think this will make both lemmas much more difficult to use. !!!